### PR TITLE
use `=` register in insert mode

### PR DESCRIPTION
**Pull request by QiBaobin**
to respect dot command

### CODE REVIEW

File: lua/registers.lua [hunk 1]
```diff
@@ -272,27 +272,15 @@ local function apply_register(register)
...
-			local key = vim.api.nvim_replace_termcodes("<c-r>", true, true, true)
+		-- Get the proper keycode for <C-R>
+		local key = vim.api.nvim_replace_termcodes("<c-r>", true, true, true)
 
-			-- Opening the expression register in normal mode magically works
-			-- I have no idea why..
+		if register == "=" then
 			vim.api.nvim_feedkeys(key .. "=", "n", true)
+		else
+			local submit = vim.api.nvim_replace_termcodes("<CR>", true, true, true)
```
Comment (tversteeg) [inline lua/registers.lua:281]: Can you add comments describing what it does similar to the rest?

**Review by tversteeg — CHANGES_REQUESTED** [review]

**Comment by tversteeg** [comment]
Thanks for the PR!

File: lua/registers.lua [hunk 2]
```diff
@@ -272,27 +272,15 @@ local function apply_register(register)
...
+			vim.api.nvim_feedkeys(key .. "=@" .. register .. submit, "n", true)
```
Comment (tversteeg) [inline lua/registers.lua:282]: This also overwrites the `=` register with `@` when using it, would it be possible to get the previous value from the `=` register first and then set it after this call?

**Review by tversteeg — CHANGES_REQUESTED** [review]

File: lua/registers.lua (hunk 2 above)
Comment (QiBaobin) [reply lua/registers.lua:282]: Good catch, :)

**Comment by QiBaobin** [comment]
> Thanks for the PR! It simplifies it a lot but unfortunately it doesn't fix the dot command..

Really, I tested on my local before the PR. could you provide your steps?

File: lua/registers.lua (hunk 1 above)
Comment (QiBaobin) [reply lua/registers.lua:281]: sure

**Comment by tversteeg** [comment]
> Really, I tested on my local before the PR. could you provide your steps?

I think I made a mistake somewhere, when I test it again now I don't see a discrepancy.

File: lua/registers.lua [hunk 3]
```diff
@@ -272,27 +272,21 @@ local function apply_register(register)
...
-			local key = vim.api.nvim_replace_termcodes("<c-r>", true, true, true)
+		-- Get the proper keycode for <C-R>
+		local key = vim.api.nvim_replace_termcodes("<c-r>", true, true, true)
 
-			-- Opening the expression register in normal mode magically works
-			-- I have no idea why..
+		if register == "=" then
 			vim.api.nvim_feedkeys(key .. "=", "n", true)
+		else
+			local old_expr_content = vim.fn.getreg('=', 1)
```
Comment (tversteeg) [inline lua/registers.lua:281]: With neovim 0.4.4 I now get `attempt to index field 'fn' (a nil value)`

**Review by tversteeg — CHANGES_REQUESTED** [review]

**Review by tversteeg — APPROVED** [review]

**Pull request merged**